library/src/hcc_detail/hipsparse.cpp

```diff
@@ -5490,6 +5500,488 @@ hipsparseStatus_t hipsparseZcsrilu02(hipsparseHandle_t         handle,
                            pBuffer));
 }
 
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+hipsparseStatus_t
+    hipsparseXbsric02_zeroPivot(hipsparseHandle_t handle, bsric02Info_t info, int* position)
+{
+    // Obtain stream, to explicitly sync (cusparse bsric02_zeropivot is blocking)
```

Comment:
Did you verify that cusparse is actually blocking?

Reply:
Their documentation says cusparseXbsric0_zeroPivot is blocking as they call cudaDeviceSynchronize but I did not verify it.